junit-platform-engine/src/main/java/org/junit/platform/engine/TestDescriptor.java

```diff
@@ -172,6 +176,40 @@ default Set<? extends TestDescriptor> getDescendants() {
 	 */
 	void removeFromHierarchy();
 
+	/**
+	 * Order all children from this descriptor.
+	 *
+	 * <p>The {@code orderer} is provided a modifiable list of child test descriptors in
+	 * this test descriptor. Never {@code null}. The {@code orderer} must return
+	 * a list of descriptors in any order, never {@code null}.
```

Comment:
I think the API would be clearer if the `ordered` received an unmodifiable list and returned the new list. Thoughts?

Reply:
It has pro's and cons.

To use `Collections.shuffle(children)` or `children.sort(...)` every orderer would have to create a new list first. Which I imagine would be quite annoying when implementing an orderer.

But the `TestDescriptor` could optimize by not doing anything if the orderer returns the same list (by reference).

Reply:
I agree that this API is "easiest to work with" and have pushed a commit that documents how to use it in the method's Javadoc.